helix-term/src/application.rs

```diff
@@ -470,7 +467,25 @@ impl Application {
                     .show_cursor(CursorKind::Block)
                     .ok();
                 restore_term().unwrap();
-                low_level::emulate_default_handler(signal::SIGTSTP).unwrap();
+
+                // SAFETY:
+                //
+                // - helix must have permissions to send signals to all processes in its signal
+                //   group, either by already having the requisite permission, or by having the
+                //   user's UID / EUID / SUID match that of the receiving process(es).
+                let res = unsafe {
+                    // A pid of 0 sends the signal to the entire process group, allowing the user to
+                    // regain control of their terminal if the editor was spawned under another process
+                    // (e.g. when running `git commit`).
+                    libc::kill(0, signal::SIGSTOP)
```

Comment:
Can you please leave a comment saying why we need this? And at the same time also mention that this could be an upstream issue. I am thinking that we can just fix it in helix for now until we figure out why this happens.

Reply:
I can add a comment that it may be an upstream issue and link to the previous issue, but I'm not sure what more you want about "why we need this" (I thought that "allowing the user to regain control of their terminal if the editor was spawned under another process (e.g. when running `git commit`)." was pretty good :sweat_smile: ). Could you give a suggestion on what's missing?

Reply:
Why we need this as in

> Currently SIGSTOP only stops main parent process (or thread, I am not sure) of helix but subsequent processes (or threads?) does not receive the signal from tokio, and the terminal will get stuck in raw mode rendering it unusable when it was stopped from a 3rd party process like git as subsequent processes are still running, we believe this could be an issue in signal-hook-tokio but the author of signal-hook says it could be a tokio issue instead. So we have to send a SIGSTOP to the entire process group to make sure it is fully stopped.

I think the explanation I have isn't clear and can be improved, but that's what I have in mind.

Reply:
Ah, I see. I'll noodle and see if I can't make it more clear. Thanks for the tip!

Reply:
OK, I've added a comment hopefully explaining more why this is necessary.